### ISSUE TITLE
maxspeed:conditional is not considered

### ISSUE THREAD

**Issue by SamuelBrucksch** [valhalla/valhalla#4553]
I just checked `maxspeed` property of an edge for a road with conditional speed limit, and it seems to ignore the `maxspeed:conditional` property.

Then i checked the lua file and it indeed only uses the `maxspeed` tag.

It's however not as easy as to replace it in lua, as conditional speed limits usually mean parsing the time and date, when they should be applied.

For me it would also be fine, to only get it as metadata, like the string in a property like `speedLimit_raw`, which contains the tag value, which is not processed, as we only need it to display the current speed limit, it shouldn't affect the actual planning too much anyways.

Would it be easily possible, to just add an additional property to the edge data, or would that rather require a bigger change? Is it maybe possible, to already attach additional data somewhow? Would it be possible, for example to replace an existent value, which is not so important, to contain the data we need, so the effort implementation wise is not that high?

What do you think would b the easiest way to do it?

Apart from that, it could actually also be handled by Valhalla, when a time is passed, valhalla could check if the time of arrival at that edge is within the conditional time frame or not and use the conditional speed limit instead then.

Here is a sample way, which has such a conditional speed limit:
https://www.openstreetmap.org/way/25575714

**Comment by nilsnolde** [comment]
it could be handled the same as the other conditional `AccessRestrictions` (though it's not one but that's what the structure is called). IRC there's plenty of "space" and those tags are so sparse, no problem adding them I guess  (it's doing binary search to retrieve access restrictions). that would at least make them part of the graph and could be retrieved by `trace_attributes`/`locate` or so. I might not think of some gotcha though.

it could also be used in costing, don't see a big problem there off the top of my head. 

in summary, all the bits & pieces to fairly easily add them are already there (specifically parsing the time component), you'll just need to follow the code paths around access restrictions.
